### PR TITLE
v2.9.1: Fixed configuration bug on single-store-mode

### PR DESCRIPTION
**Pull request by pini-girit**
[!!!] Not fully tested yet.

* Fixed configuration bug on single-store-mode.
* Removed scopes & restore checkbox from store-only configuration fields.

### CODE REVIEW

File: Observer/Config/Save.php [hunk 1]
```diff
@@ -65,40 +65,58 @@ public function execute(Observer $observer)
             $this->cacheTypeList->cleanType(Config::TYPE_IDENTIFIER);
             $this->appConfig->reinit();
 
-            $scope = $scopes = null;
+            $scope = $scopes = \Magento\Framework\App\Config\ScopeConfigInterface::SCOPE_TYPE_DEFAULT;
```
Comment (sharp80) [inline Observer/Config/Save.php]: why do we need 2 different variables: "scope" and "scopes"? i do not see any different value in this function between the two - it looks redundant to me. Am I missing something?

File: Model/Config.php [hunk 2]
```diff
@@ -286,23 +295,25 @@ public function isActivated($scopeId = null, $scope = null)
...
+     * @param  string|null                     $scopes
      * @return $this
      */
-    public function setStoreCredentialsAndIsEnabled($appKey, $secret, $isEnabled, $storeId = null)
+    public function setStoreCredentialsAndIsEnabled($appKey, $secret, $isEnabled, $storeId = null, $scopes = ScopeInterface::SCOPE_STORES)
     {
-        $this->resourceConfig->saveConfig(self::XML_PATH_YOTPO_APP_KEY, $appKey, ScopeInterface::SCOPE_STORES, $storeId ?: $this->storeManager->getStore()->getId());
-        $this->resourceConfig->saveConfig(self::XML_PATH_YOTPO_SECRET, ($secret ? $this->encryptor->encrypt($secret) : null), ScopeInterface::SCOPE_STORES, $storeId ?: $this->storeManager->getStore()->getId());
-        $this->resourceConfig->saveConfig(self::XML_PATH_YOTPO_ENABLED, $isEnabled, ScopeInterface::SCOPE_STORES, $storeId ?: $this->storeManager->getStore()->getId());
+        $this->resourceConfig->saveConfig(self::XML_PATH_YOTPO_APP_KEY, $appKey, $scopes, is_null($storeId) ? $this->storeManager->getStore()->getId() : $storeId);
```
Comment (sharp80) [inline Model/Config.php]: please do not use one liners. 
change the expression:
   is_null($storeId) ? $this->storeManager->getStore()->getId() : $storeId
to:

if (is_null($storeId))
    $storeId  = $this->storeManager->getStore()->getId();

please put the above expression in the beginning of the function instead of embedding it inside the saveConfig function.

File: view/adminhtml/templates/system/config/launch_yotpo_button.phtml [hunk 3]
```diff
@@ -6,6 +6,7 @@
 <?php if (!$block->isStoreScope()) : ?>
     <p class="note"><?= __('Yotpo can only be connected, enabled, or disabled via the Store View menu.<br>To connect your App Key and API secret, first select the relevant Store View.') ?></p>
     <p></p>
+    <!--style>tr#row_yotpo_settings_active,tr#row_yotpo_settings_app_key,tr#row_yotpo_settings_secret{display:none!important;}</style-->
```
Comment (sharp80) [inline view/adminhtml/templates/system/config/launch_yotpo_button.phtml]: do not add unneeded comments.

File: Observer/Config/Save.php [hunk 4]
```diff
@@ -65,40 +65,58 @@ public function execute(Observer $observer)
...
+                $this->resetStoreCredentials($scopeId, $scopes);
                 return true;
             }
+
             //Check if appKey is unique:
             if ($appKey) {
-                foreach ($this->yotpoConfig->getAllStoreIds(false) as $key => $storeId) {
-                    if ($storeId !== $scopeId && $this->yotpoConfig->getAppKey($storeId) === $appKey) {
-                        $this->yotpoConfig->resetStoreCredentials($scopeId);
+                foreach ($this->yotpoConfig->getAllStoreIds() as $key => $storeId) {
```
Comment (sharp80) [inline Observer/Config/Save.php]: why did you change this logic? it has nothing to do with single store mode - single store mode has only 1 store - in this logic looks for 2 different stores with the same app key. 

also - why did you remove the false parameter from the getAllStoresIds()?

File: Console/Command/ResetCommand.php [hunk 5]
```diff
@@ -108,7 +108,7 @@ private function resetConfig()
     {
         $this->resourceConnection->getConnection()->delete(
             $this->resourceConnection->getTableName('core_config_data'),
-            "path LIKE '" . \Yotpo\Yotpo\Helper\Data::XML_PATH_ALL . "/%'"
+            "path LIKE '" . \Yotpo\Yotpo\Model\Config::XML_PATH_YOTPO_ALL . "/%'"
```
Comment (sharp80) [inline Console/Command/ResetCommand.php]: what is the reason for this change? what is the connection to single store mode?

File: Model/Config.php (hunk 2 above)
Comment (pini-girit) [reply Model/Config.php]: ok

File: Observer/Config/Save.php (hunk 4 above)
Comment (pini-girit) [reply Observer/Config/Save.php]: false is the default, so no need for it.

File: Console/Command/ResetCommand.php (hunk 5 above)
Comment (pini-girit) [reply Console/Command/ResetCommand.php]: It's a bug that I found during the single-store issue (actually I found 2 bugs & solved them on a different commit).

File: Observer/Config/Save.php (hunk 1 above)
Comment (pini-girit) [reply Observer/Config/Save.php]: They are not always the same.

File: view/adminhtml/templates/system/config/launch_yotpo_button.phtml (hunk 3 above)
Comment (pini-girit) [reply view/adminhtml/templates/system/config/launch_yotpo_button.phtml]: It's was a mistake on the commit, I pushed a fix few minutes after that...

File: Observer/Config/Save.php (hunk 1 above)
Comment (sharp80) [reply Observer/Config/Save.php]: please elaborate - where in the code can these 2 be different?

File: composer.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
     "name": "yotpo/module-review",
     "description": "Yotpo Reviews extension for Magento2",
-    "version": "2.9.0",
+    "version": "2.9.1",
```
Comment (sharp80) [inline composer.json]: why was the version changed to 2.9.1?
